prog/dftb+/lib_timedep/linrespcommon.F90

```diff
@@ -227,36 +227,66 @@ end subroutine rindxvv
 
   !> Builds array to convert from orbital pairs to a compound index, reverse of index generated by
   !> getSPExcitations (which were then potentially windowed afterwards)
-  subroutine rindxov_array(win, lumo, nxov, getij, iatrans)
+  subroutine rindxov_array(win, nxov, nxoo, nxvv, getia, getij, getab, iatrans)
 
     !> array for indexing excitations
     integer, intent(in) :: win(:)
 
-    !> lowest occupied molecular orbital
-    integer, intent(in) :: lumo
-
     !> Number of transitions from occupied to virtual
     integer, intent(in) :: nxov
 
+    !> Number of transitions from occupied to occupied
+    integer, intent(in) :: nxoo
+
+    !> Number of transitions from virtual to virtual
+    integer, intent(in) :: nxvv
+
     !> array of the occupied->virtual pairs
+    integer, intent(in) :: getia(:,:)
+
+    !> array of the occupied->occupied pairs
     integer, intent(in) :: getij(:,:)
 
+    !> array of the virtual->virtual pairs
+    integer, intent(in) :: getab(:,:)
+
     !> resulting index array from orbital pairs to compound index
-    integer, intent(out) :: iatrans(:,lumo:,:)
+    integer, intent(out) :: iatrans(:,:,:)
 
-    integer :: ia
+    integer :: ia, ij, ab
 
+    @:ASSERT(size(getia,dim=2) == 3)
     @:ASSERT(size(getij,dim=2) == 3)
+    @:ASSERT(size(getab,dim=2) == 3)
```

Comment:
I think, we can leave away those asserts. (The compilers bounday checks should catch this error in debug mode automatically.)